### PR TITLE
Partial OS X build fix

### PR DESCRIPTION
**Pull request by droark**
- Arrange for Makefiles to properly detect the Clang/OSX combo.
- Update OSX build script.

OSX build still fails.

### CODE REVIEW

File: cppForSwig/cryptopp/configure.ac [hunk 1]
```diff
@@ -50,7 +52,13 @@ AC_CHECK_TYPES([ptrdiff_t])
...
 AC_CHECK_FUNCS([gethostbyname gettimeofday memmove memset pow select setlocale socket])
 
-AM_CONDITIONAL([CRYPTOPP_HAVE_CLANG], [test $CXX == clang++])
+case $host in
+  *darwin*)
+  BUILD_OS=darwin
+  ;;
+esac
+
+AM_CONDITIONAL([CRYPTOPP_HAVE_CLANG], [test x$BUILD_OS = xdarwin])
```
Comment (droark) [inline cppForSwig/cryptopp/configure.ac]: Yeah, I was going to clean that up later. I can fix it now.

Regarding GCC and OS X, I think you can build but a vast majority use Xcode, which defines GCC as a front for Clang (i.e., gcc = clang, g++ = clang++, etc.).

**Comment by droark** [comment]
@goatpig - I just did a fixup. The problem is that "IS_X86" is failing. That needs to be revised. I think it'll be fine after that.

**Comment by goatpig** [comment]
That uses the g++ __builtin__ introduces starting 4.8

The idea is that it's trying to figure out which x86 extended instruction set the CPU supports. The underlying code defines these method otherwise, which will conflict at compile time since the compiler already supports those.

Since the macro is locked to GCC, I think your best path of action is to find another m4 dedicated to clang.

**Comment by droark** [comment]
@goatpig - Assuming I understand you correctly, it shouldn't be necessary to use that method to determine if X86 is present; simply checking the build machine type will suffice. Once I did that, the code worked. Please take a look now and let me know if I broke anything in the process. (Well, RPi cross-compiles may be broken, but that's probably not due solely to any changes I made!)

**Comment by goatpig** [comment]
Your change breaks compilation on plain Linux. The gcc_x86 m4 macro does not simply detect the architecture, it enumerates the feature sets and populates X86_FEATURE_CFLAGS with the proper compiler flag for GCC.

Keep in that these features, albeit on the silicon, have to be exposed by the compiler. You need that detection for GCC. You can if else around the use of that flag for clang as you see fit, but you can't remove it as a whole.

**Comment by droark** [comment]
@goatpig - Restored the code in question (and added a couple of M4 files that were supposed to be included but were never placed in the repo). Jut did a build on an Ubuntu VM. Looks good to me. It is choking on a tr() call but that's a different story. I'll mention this and another issue I found on the forum.

**Pull request merged**